cmd/git/branch.go

```diff
@@ -0,0 +1,74 @@
+package git
+
+import (
+  "fmt"
+  "io/ioutil"
+  "log"
+  "strings"
+
+  "github.com/Originate/gt/cmd/util"
+)
+
+
+func EnsureDoesNotHaveBranch(branchName string) {
+  if HasBranch(branchName) {
+    util.ExitWithErrorMessage(fmt.Sprintf("A branch named '%s' already exists", branchName))
+  }
+}
+
+
+func GetCurrentBranchName() string {
+  if IsRebaseInProgress() {
+    filename := fmt.Sprintf("%s/.git/rebase-apply/head-name", GetRootDirectory())
+    rawContent, err := ioutil.ReadFile(filename)
+    if err != nil {
+      log.Fatal(err)
+    }
+    content := strings.TrimSpace(string(rawContent))
+    return strings.Replace(content, "refs/heads/", "", -1)
+  } else {
+    return util.GetCommandOutput([]string{"git", "rev-parse", "--abbrev-ref", "HEAD"})
+  }
+}
+
+
+func GetTrackingBranchName(branchName string) string {
+  return "origin/" + branchName
+}
+
+
+func HasBranch(branchName string) bool {
+  cmd := []string{"git", "branch", "-a"}
+  lines := strings.Split(util.GetCommandOutput(cmd), "\n")
+  for _, line := range(lines) {
```

Comment:
In conjunction with inlining cmd I feel like that gets out of hand. I updated this to using a temp variable output and then inline lines as just splitting that by newline.